### PR TITLE
Introduce batch application state and rename BatchJobSubmission to SubmitBatchApp

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Introduce batch application submission state.

The initialized state is `PENDING`, when it gets the applicationStatus at first time, it should update the status to `RUNNIGN`.

And when the batch operation terminates, update the final state to `FINISHED` or `FAILED` or `KILLED`.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchAppSubmissionState.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.operation
+
+object BatchAppSubmissionState extends Enumeration {
+  type BatchAppSubmissionState = Value
+
+  val SUBMITTING, RUNNING, FINISHED, FAILED, KILLED = Value
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchAppSubmissionState.scala:23]: SUBMITTING -> PENDING?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchAppSubmissionState.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.operation
+
+object BatchAppSubmissionState extends Enumeration {
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchAppSubmissionState.scala:20]: ApplicationState?

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchAppSubmissionState.scala:20]: it is not absolute same with the state of ApplicationReport.

So, I am thinking how to name it.

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchAppSubmissionState.scala:20]: For example, the `BatchAppSubmissionState` is FINISHED and the application state is `RUNNING`.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchAppSubmissionState.scala (hunk 1 above)
Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchAppSubmissionState.scala]: updated

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchAppSubmissionState.scala (hunk 2 above)
Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchAppSubmissionState.scala:20]: we shall call it submitted...

**Comment by yaooqinn** [comment]
OperationState seems can cover all scenarios of submission and app state

**Comment by turboFei** [comment]
> OperationState seems can cover all scenarios of submission and app state

I see.

**Pull request closed without merging**